src/App.js

```diff
@@ -1,25 +1,99 @@
-import logo from './logo.svg';
-import './App.css';
+import { useState } from "react";
+import "./App.css";
+import contactsJson from "./contacts.json"
 
 function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+  const [contacts, setContacts] = useState(contactsJson.slice(0, 5))
+  const [remainingContacts, setRemainingContacts] = useState(contactsJson.slice(5))
+
+  const handleAddRandomContact = () => {
+    const numRemainingContacts = remainingContacts.length
+    if (numRemainingContacts === 0){
+      return
+    }
+    const randomIndex = Math.floor(Math.random() * numRemainingContacts)
+    setContacts(prevContacts => {
+      const newContacts = [...prevContacts]
+      newContacts.push(remainingContacts[randomIndex])
+      return newContacts
+    })
+    setRemainingContacts(prevRemainingContacts => {
+      const newRemainingContacts = [...prevRemainingContacts]
+      newRemainingContacts.splice(randomIndex, 1)
+      return newRemainingContacts
+    })
+  }
```

Comment:
Great solution!

Alternative approach, would be to have only one state for `contacts`, and then check remaining contacts as you go, using `filter()`. 

First you want to create an array where you take list from database and remove all the contacts already displayed. And get random contact from it instead

```
  const addRandomContact = ()=>{
    setContacts((prevContacts)=>{
      const newContactsList = contactsArr.filter((contact)=> {
        return !prevContacts.includes(contact);
      });
      const getRandomContact = newContactsList[Math.floor(Math.random()* newContactsList.length)];
      const newConctactSelection = [...prevContacts,getRandomContact]
      return newConctactSelection;
    })
  };
```